M3-3599 Longview e2e tests

jstamis-linode opened the pull request:
Longview UI tests

* Created test for the longview plan subscription and longview clients.
* All tests can be run individually or together. The beforeEach and API calls will set up the data for the tests to run and use the information
from the responses for verification.

**note** there is an ugly browser.pause(1500) in the changePlan function.
This is because once the button is selected it is set to a disabled mode
so the check for the button to not be enabled will not work. Need a better
way to wait for the loading state on the button.

When running tests remember to use a user that does NOT have longview
clients installed. The tests will clear out all longview clients to make
sure the tests run reliably.

Added in some needed selectors to allow for UI automation to work easier. We are now using the data-testid instead of data-qa for testing attribiutes.

Fixes for crumb/header changes
changed title selector from '[data-qa-dialog-title]' to 'h1'

## Type of Change
- Non breaking change ('update', 'change')

## Applicable E2E Tests

To run relevant E2E tests, run these commands in 3 separate terminals:

1. `yarn up`
2. `yarn selenium`
3. `yarn e2e --file longview/longview-plan-details.spec.js`

## Note to Reviewers

When running tests remember to use a user that does NOT have longview
clients installed. The tests will clear out all longview clients to make
sure the tests run reliably.

johnwcallahan commented on packages/manager/e2e/specs/longview/longview-plan-details.spec.js:
```diff
@@ -0,0 +1,115 @@
+      expect(LongviewLanding.noLVClients.$('p').getText())
+        .withContext(`${assertLog.incorrectText}`)
+        .toBe(`${longviewText.noClients}`);
+      expect(LongviewLanding.upgradeText.getText())
```
At first I was getting errors because I already had LV Pro, so the upgradeText wasn't there. We should probably call `setPlanLVFree()` in a `beforeAll()`. 

(I was also a Managed account, which gives you LV Pro, so we should account for that too).

jstamis-linode replied:
Yeah, this one is challenging. As to the managed issue I'm not sure how we can get that to work as you cannot reset a managed user. If you have any ideas on that one let me know. If I can reset a user from being managed I will gladly do that. I will change the reset with a before all as well.

johnwcallahan replied:
Good point about not being about to reset a Managed account. We'll probably have to do with a warning for now.

jstamis-linode commented:
I fixed a small issue with a merge from Pat that overwrote my header changes. I also added a slightly longer wait time for the client to open. If you are still having issues with a hanging browser could we do a session together to see what might be different? I have asked @Jskobos to pull this branch again and see if he has any issues. I still have not had any hanging or failing issues.

Jskobos commented:
@jstamis-linode @johnwcallahan I can still run these consistently using prod-test-011 (empty/unused account) without issue

johnwcallahan commented:
@jstamis-linode It could have been a freak thing. If everyone else is seeing passing tests, and you're seeing passing tests on my test account, I feel okay about merging.
